baseplate.go

```diff
@@ -124,11 +149,14 @@ func Serve(ctx context.Context, server Server) error {
 			// Initialize a channel to pass the result of server.Close().
 			closeChannel := make(chan error)
 
-			// Tell the server to close.
+			// Tell the server and any provided closers to close.
 			//
 			// This is a blocking call, so it is called in a separate goroutine.
 			go func() {
-				closeChannel <- server.Close()
+				bc := batchcloser.New(preShutdown...)
+				bc.Add(server)
+				bc.Add(postShutdown...)
```

Comment:
I'm perhaps naively relying on the assumption that `preShutdown` and `postShutdown` will work appropriately if unset (presumably either `nil` or an empty slice, and in either case not problematic for `batchcloser.New` or `batchcloser.Add`.

Reply:
I think this should be fine.

Reply:
Same style comment as above for this line:

```go
				bc.Add(args.postShutdown...)
```